### ISSUE TITLE
Modifiers pressed before 'TempSwitchLayerN' not being released

### ISSUE THREAD

**Issue by Sinitax0** [pixeltris/GK6X#66]
**Expected Behavior**: If e.g LWin (standard mapping) is pressed before switching to another layer via a TempSwitchLayerN key, the modifier (WIN) should be released, when the key (LWin) is released.

**Observed Behavior**: The modifier is not released and continues to apply for other keypresses until the modifier is pressed again, to send the 'release' event I assume. This only happens when the keypresses occur in the following order: MOD(DOWN), TEMP_LAYER(DOWN), MOD(UP), TEMP_LAYER(UP)  

For reference, I have a GK61 with optical red switches, ID 655491117.

**Comment by pixeltris** [comment]
I guess the way this is implemented is done on a per-firmware basis. Here is what happens on a couple of other keyboards:

GK84: The modifier key is held down until released (using the expected behavior you described above).
GK68: The modifier key is released as soon as the temp switch layer is press down.

Unfortunately I have no control over the firmware so this is something you'd have to take up with the keyboard manufacturer (who will most likely ignore you unfortunately).

**Comment by Sinitax0** [comment]
Thank you for the fast answer, I sort of suspected this was the case.

I guess my only option is to modify the firmware via something similar to: https://github.com/wgwoods/gk64-python

**Comment by pixeltris** [comment]
Yea, it would be nice for QMK to be supported on GK6X boards, but there are so many one them, just supporting one would be quite some effort.

**Comment by Sinitax0** [comment]
I actually managed to get this working after realizing what was causing this behavior.

When releasing a key during a temp layer switch the firmware actually sends the release event of the key in the temp layer, instead of that of the key for which a press event was sent previously. I was disabling every key on every layer before mapping my keys, because I wanted to be confident in the output of each key. This caused the modifiers in non-base layers to be disabled, which led to the *observed behavior* described above. By not disabling these modifiers (having the same keys mapped in the layer you're switching to) I arrived at my *expected behavior*.

This actually (atleast for my keyboard) isn't even an issue for non-modifier keys, since these are released automatically upon entering another layer (very neat!). As such, there are still plenty of keys in other layers which can be remapped without causing unexpecteds behavior.
